app/experimenter/docs/adrs/0001-create-new-nimbus-application.md

```diff
@@ -0,0 +1,119 @@
+  - [Current Experimenter codebase may not align with FXA tools/processes/practices](https://github.com/mozilla/fxa/blob/main/docs/adr/0020-application-architecture.md)
+  - Somewhat constricted by core designs decisions that were made in service of existing Normandy experimentation workflows that may not apply to Nimbus
+
+### Start A New Application In Python/Django
+
+Creating a new web application in Python/Django would allow us to reuse much of the existing code/infrastructure where applicable while clearly isolating the concerns of supporting existing experiment types and building out new Nimbus functionality.
+
+- Pros
+  - Start by cloning the repo and deleting all the unnecessary UI/DB/APIs
+  - Get a big head start by not needing to reimplement the reusable parts
+  - Isolate Nimbus concerns without fear of breaking Normandy experiments
+  - Consistent development and deployment workflow for both old and new applications
+  - Opportunity to rethink core code/architectural/tooling decisions where appropriate
+- Cons
+  - Lose single source of truth
+    - Possibly require people to know which application to refer to depending on experiment type
+    - Possibly create an API proxy in front of both to have a single endpoint for dependent callers
+    - More difficult to correlate experiments/learnings across the two applications
+  - Changes that must apply to both Normandy and Nimbus experiments must be made in two places while they're both being actively maintained
+  - Divergence in tooling/processes over time may create confusion when context switching between the two
+  - Possibly still somewhat constricted by past core design decisions
+  - [Python/Django may not align with FXA tools/processes](https://github.com/mozilla/fxa/blob/main/docs/adr/0020-application-architecture.md)
+
+### Start A New Application in TypeScript/Node
+
+Creating a new web application in TypeScript/Node would allow us to completely restart our thinking about the entire structure of the application which would result in both code and workflow/UI that are purpose built from the ground up to solely address the concerns of Nimbus experiments.
+
+- Pros
+  - Totally greenfield approach allows fresh perspectives to flourish
+  - Stronger sense of collective ownership from all new contributors
+  - Every decision and line of code built specifically to address the needs of Nimbus
+  - TypeScript provides excellent feedback and guarantees which improves development experience
+  - Native async/websocket support allows richer UI experience when coupled with SPA frameworks (like React)
+  - [Fits more cohesively into the FXA team's tooling/processes](https://github.com/mozilla/fxa/blob/main/docs/adr/0020-application-architecture.md)
+  - Logic/types in TS can be easily shared across multiple projects potentially including Desktop/Mobile clients
+  - TS may integrate more smoothly with Rust than Python does to more tightly couple the server and the new Experiment Client
```

Comment:
At first read, I don't buy this "they both have strong typesystems and hence might couple more smoothly" suggestion; do we have a concrete example of what sort of benefit we might derive from this in practice? The systems are going to be intermediated by an webapp API on the server (http+json? graphql?) and by a foreign-language-bindings layer on the client which might negate any coupling benefits here.

(I can imagine a benefit of having *similarly-structured* datatypes on the server and client, but this bullet seems to imply a tighter level of coupling than that)

Reply:
Well I was thinking of the case where we have the Rust experiments client, and we want to strongly enforce the contract between that client and the server by including the client in the server's CI test flow.  You can [expose Rust functions to Python](https://github.com/PyO3/pyo3), and you can [expose Rust functions to TypeScript](https://github.com/rustwasm/wasm-bindgen), but with TypeScript it'll also [generate the TypeScript stubs from the Rust types](https://rustwasm.github.io/wasm-bindgen/reference/cli.html#--typescript) so I think that'll allow tighter integration and a smoother developer workflow, but I'm purely speculating.

Reply:
> we have the Rust experiments client, and we want to strongly enforce the contract between that client
> and the server by including the client in the server's CI test flow

Oh, interesting point, thanks!

Amusing semi-aside: the currently plan for exposing the Rust SDK to other languages is to use a [cross-language bindings generator](https://github.com/mozilla/uniffi-rs), which coincidentally happens to [already support python](https://github.com/mozilla/uniffi-rs/tree/main/uniffi_bindgen/src/bindings/python) but for which we haven't tackled JS/TS yet.

Reply:
Personal opinion of course, but I think I'd prefer to (continue to) have a well-defined JSON-based protocol rather than relying on any language magic to connect the client and the server.  If we want to run integration tests, we can make a cli client based on the Rust library and puppet it as its own process.